## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-tls.sh | :x: [more...](#fbf897c2-61a5-4f99-aae0-04d6d98809e7) |
| mint-gateway-s3.sh | :x: [more...](#e5368d84-ace7-43ac-a813-5eba00002d0b) |
| mint-gateway-azure.sh | :x: [more...](#c2b01360-a055-4e17-8fe4-55e4de89eeed) |
<a name="c2b01360-a055-4e17-8fe4-55e4de89eeed"></a>
### 6418-818de1e/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32466
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d070e74198f1:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 2 minutes and 49 seconds
(2/13) Running aws-sdk-go tests ... done in 2 seconds
(3/13) Running aws-sdk-java tests ... done in 5 seconds
(4/13) Running aws-sdk-php tests ... done in 40 seconds
(5/13) Running aws-sdk-ruby tests ... done in 44 seconds
(6/13) Running mc tests ... done in 7 minutes and 28 seconds
(7/13) Running minio-dotnet tests ... done in 6 minutes and 9 seconds
(8/13) Running minio-go tests ... FAILED in 7 minutes and 38 seconds
{
  "args": {
    "policy": "{\"expiration\":\"2018-09-15T14:32:11.658Z\",\"conditions\":[[\"eq\",\"$bucket\",\"minio-go-test-w3peqvubh01vmeqi\"],[\"eq\",\"$key\",\"y0dxndz3tkmm0wx5sb4pw3viw2ador\"],[\"eq\",\"$Content-Type\",\"binary/octet-stream\"],[\"eq\",\"$x-amz-meta-4s4os4zekm0nx6h4fsikqsne3gkvf3\",\"s0r2yyun1fqlirsi1p9lqumjlo2avm\"],[\"content-length-range\", 10, 1048576]]}"
  },
  "duration": 371,
  "error": "400 Bad Request",
  "function": "PresignedPostPolicy(policy)",
  "message": "Http request failed",
  "name": "minio-go: testPresignedPostPolicy",
  "status": "FAIL"
}

Executed 7 out of 13 tests successfully.
```
<a name="fbf897c2-61a5-4f99-aae0-04d6d98809e7"></a>
### 6418-818de1e/mint-tls.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         1
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d5b3c8ea7bb3:/mint/log /tmp/mint-logs'
(1/13) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-4c2sqjoeycy93b",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 318,
  "error": "RequestError: send request failed\ncaused by: Put https://localhost/aws-sdk-go-test-4c2sqjoeycy93b: dial tcp 127.0.0.1:443: getsockopt: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="e5368d84-ace7-43ac-a813-5eba00002d0b"></a>
### 6418-818de1e/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:30918
ACCESS_KEY:           AKIAJ3ORETMQUQXBBWGA
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e347b604e4d3:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 3 minutes and 5 seconds
(2/13) Running aws-sdk-go tests ... done in 4 seconds
(3/13) Running aws-sdk-java tests ... done in 6 seconds
(4/13) Running aws-sdk-php tests ... done in 1 minutes and 25 seconds
(5/13) Running aws-sdk-ruby tests ... done in 1 minutes and 0 seconds
(6/13) Running mc tests ... done in 7 minutes and 48 seconds
(7/13) Running minio-dotnet tests ... done in 8 minutes and 6 seconds
(8/13) Running minio-go tests ... done in 9 minutes and 58 seconds
(9/13) Running minio-java tests ... done in 16 minutes and 40 seconds
(10/13) Running minio-js tests ... done in 1 minutes and 31 seconds
(11/13) Running minio-py tests ... FAILED in 30 minutes and 6 seconds
{
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 600, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 384, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 2, in raise_from\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 380, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.5/http/client.py\", line 1197, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.5/http/client.py\", line 297, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.5/http/client.py\", line 266, in _read_status\n    raise RemoteDisconnected(\"Remote end closed connection without\"\nhttp.client.RemoteDisconnected: Remote end closed connection without response\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 165, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.5/dist-packages/minio/api.py\", line 286, in make_bucket\n    headers=headers)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/poolmanager.py\", line 322, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 638, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/util/retry.py\", line 398, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minikube', port=30918): Max retries exceeded with url: /minio-py-test-1c849f54-b864-4e4a-897f-0c9ad43debe2/ (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1707, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 173, in test_make_bucket_default_region\n    raise Exception(err)\nException: HTTPConnectionPool(host='minikube', port=30918): Max retries exceeded with url: /minio-py-test-1c849f54-b864-4e4a-897f-0c9ad43debe2/ (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n",
  "duration": 1806020,
  "args": {
    "bucket_name": "minio-py-test-1c849f54-b864-4e4a-897f-0c9ad43debe2",
    "location": "default value ('us-east-1')"
  },
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "status": "FAIL",
  "message": "HTTPConnectionPool(host='minikube', port=30918): Max retries exceeded with url: /minio-py-test-1c849f54-b864-4e4a-897f-0c9ad43debe2/ (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))"
}

Executed 10 out of 13 tests successfully.
```

--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -271,7 +271,7 @@ func serverMain(ctx *cli.Context) {
 				logger.Info(`Supplying credentials from your 'config.json' is **DEPRECATED**, Access key and Secret key in distributed server mode is expected to be specified with environment variables MINIO_ACCESS_KEY and MINIO_SECRET_KEY. This approach will become mandatory in future releases, please migrate to this approach soon.`)
 			} else {
 				// Credential is not available anywhere by both means, we cannot start distributed setup anymore, fail eagerly.
-				logger.Fatal(uiErrEnvCredentialsMissingServer(nil), "Unable to initialize distributed Minio server.")
+				logger.Fatal(uiErrEnvCredentialsMissingDistributed(nil), "Unable to initialize the server in distributed mode")
 			}
 		}
 	}

--- a/cmd/ui-errors.go
+++ b/cmd/ui-errors.go
@@ -74,14 +74,14 @@ Secret key should be in between 8 and 40 characters.`,
 
 	uiErrEnvCredentialsMissingGateway = newUIErrFn(
 		"Credentials missing",
-		"Please provide correct credentials",
-		`Access key and Secret key should be specified in Gateway mode from environment variables MINIO_ACCESS_KEY and MINIO_SECRET_KEY respectively.`,
+		"Please set your credentials in the environment",
+		`In Gateway mode, access and secret keys should be specified via environment variables MINIO_ACCESS_KEY and MINIO_SECRET_KEY respectively.`,
 	)
 
-	uiErrEnvCredentialsMissingServer = newUIErrFn(
+	uiErrEnvCredentialsMissingDistributed = newUIErrFn(
 		"Credentials missing",
-		"Please provide correct credentials",
-		`Access key and Secret key should be specified in distributed server mode from environment variables MINIO_ACCESS_KEY and MINIO_SECRET_KEY respectively.`,
+		"Please set your credentials in the environment",
+		`In distributed server mode, access and secret keys should be specified via environment variables MINIO_ACCESS_KEY and MINIO_SECRET_KEY respectively.`,
 	)
 
 	uiErrInvalidErasureEndpoints = newUIErrFn(

